examples/amm/src/contract.rs

```diff
@@ -57,11 +57,11 @@ impl Contract for Amm {
 
     async fn execute_message(
         &mut self,
-        context: &MessageContext,
+        runtime: &mut ContractRuntime,
```

Comment:
Do we plan to go all the way for `runtime: &mut ContractRuntime<Self::Message>` and `runtime.schedule_message(m)`? Otherwise, perhaps `&ContractRuntime` is more logical?

Reply:
Yep, that's the goal. Also, it's single-threaded, so I don't think there's any practical difference between `&` and `&mut` here :thinking: